concurrency-limits-core/src/main/java/com/netflix/concurrency/limits/limit/functions/Log10RootFunction.java

```diff
@@ -15,18 +15,19 @@
  */
 package com.netflix.concurrency.limits.limit.functions;
 
-import java.util.function.Function;
-import java.util.stream.IntStream;
+import java.util.function.IntUnaryOperator;
 
 /**
  * Function used by limiters to calculate thredsholds using log10 of the current limit.  
  * Here we pre-compute the log10 of numbers up to 1000 as an optimization.
  */
-public final class Log10RootFunction implements Function<Integer, Integer> {
+public final class Log10RootFunction implements IntUnaryOperator {
```

Comment:
Not backwards compatible, but I'd be surprised if this is widely used outside of the library itself.

Reply:
If we're going to break compatibility, can we also reduce visibility to avoid concerns in the future?

Reply:
@kilink because this is a binary incompatible change and we want to be good to our OSS folks, would you mind possibly moving the updates into a new class implementation (i.e. fork), while deprecating the existing implementations? This would allow folks time to switch over and/or not break. We can remove the deprecated classes in a release cycle or two and swap everything over to the optimized versions.

Otherwise, we will need to rev the next release as a breaking change release, which is not as OSS friendly.

Reply:
Alright, I've updated the PR to add a new class, `Log10RootIntFunction`, and reverted the original one.